### PR TITLE
FE-1325 - Toast Component

### PR DESCRIPTION
**Pull request by dawidzarzycki**
# Description

- Create new standalone component `dismiss-button`
- Update `toast` component to use `dismiss-button`.
- Update `toast` component to use `style-components`
- Update `message` component to use `dismiss-button`
- Update `message` component styling
- Update `tests` and snapshots

# Screenshots / Gifs
- Toast Button
Zrzut ekranu 2019-05-21 o 15 43 11
- Message Button
Zrzut ekranu 2019-05-21 o 15 43 23

# Testing Instructions
- Need to re-test `Message` component

### CODE REVIEW

File: src/components/dismiss-button/dismiss-button-variants.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default {
+  error: '#c7384f',
```
Comment (izakrzewska) [inline src/components/dismiss-button/dismiss-button-variants.js]: is it possible not to hardcode colors here?

File: src/components/dismiss-button/dismiss-button.spec.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+describe('when in classic mode', () => {
+  OptionsHelper.colors.forEach((variant) => {
+    describe(`when rendered as ${variant}`, () => {
+      it('should match the snapshot', () => {
+        const wrapper = render({ variant, theme: classicTheme });
+        expect(wrapper).toMatchSnapshot();
+      });
+
+      describe('when transparent prop is set to true', () => {
+        it('should apply white color background', () => {
```
Comment (izakrzewska) [inline src/components/dismiss-button/dismiss-button.spec.js]: transparent

File: src/components/toast/toast.component.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  children: PropTypes.node,
+  /** Determines if the toast is open. */
+  open: PropTypes.bool,
+  /** Callback for when dismissed. */
+  onDismiss: PropTypes.func
+};
+
+Toast.defaultProps = {
+  as: 'warning',
+  className: '',
```
Comment (izakrzewska) [inline src/components/toast/toast.component.js]: is default prop for classname necessary?

File: src/components/dismiss-button/dismiss-button-variants.js (hunk 1 above)
Comment (dawidzarzycki) [reply src/components/dismiss-button/dismiss-button-variants.js]: updated

File: src/components/dismiss-button/dismiss-button.spec.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    it('should render correctly', () => {
+      expect(shallow(<DismissButton />)).toHaveLength(1);
+    });
+  });
+
+  describe('when render with no additional props', () => {
+    OptionsHelper.messages.forEach((variant) => {
+      it(`should match snapshot for ${variant}`, () => {
+        const wrapper = render({ variant });
+        expect(wrapper).toMatchSnapshot();
```
Comment (grabkowski) [inline src/components/dismiss-button/dismiss-button.spec.js]: It's better to use `assertStyleMatch` here, as variant is responsible for a single css style

File: src/components/toast/toast.component.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        </CSSTransitionGroup>
+      </Portal>
+    );
+  }
+}
+
+Toast.propTypes = {
+  variant: PropTypes.oneOf(OptionsHelper.colors),
+  /** Customizes the appearance in a legacy theme through colour see the 'iconColorSets' for possible values) */
+  as: PropTypes.string,
```
Comment (grabkowski) [inline src/components/toast/toast.component.js]: shouldn't that be the same type as `variant`?

Comment (dawidzarzycki) [reply src/components/toast/toast.component.js]: Probably yeah, but It's an original one

File: src/components/toast/toast.component.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  toastContent() {
+    if (this.props.open) {
+      return (
+        <ToastStyle
+          variant={ this.props.variant || this.props.as }
+          className={ this.componentClasses }
+          { ...tagComponent('toast', this.props) }
+        >
+          <ToastTypeStyle variant={ this.props.variant || this.props.as }>
+            <Icon className='carbon-toast__type-icon' type={ this.props.variant || this.props.as } />
```
Comment (grabkowski) [inline src/components/toast/toast.component.js]: please change that icon to a styled component based on the `Icon` component

**Comment by benwilsonsage** [comment]
Looking good! Couple of thoughts...

New theme:
Expected: If a single row of text, toast height should be 40px.
Actual: If a single row of text, height is 49px.

New theme:
Expected: Dismiss icons are 12*12px, and slate tint 40 in colour.
Actual: Dismiss icons are green or red and larger.

New theme:
Expected: Animation follows the example at the bottom of this page (click the 'Save' button in the dialog at the bottom of the page): http://designsystem.sage.com.s3-website.eu-central-1.amazonaws.com/patterns/errors-validation/
Actual: Animation follows classic theme.

Classic theme:
Expected: Toast corners are square (no border radius).
Actual: Border radius applied.

Classic theme:
Expected: Classic colour for red results in #c7384f
Actual: Colour is #d63f40

File: src/components/message/message.component.js [hunk 7]
```diff
@@ -45,15 +45,16 @@ const Message = (props) => {
...
+  as: 'info',
   border: true,
   open: true,
   roundedCorners: true,
   transparent: false
 };
 
 Message.propTypes = {
-  messageType: PropTypes.oneOf(OptionsHelper.colors),
+  variant: PropTypes.oneOf(OptionsHelper.colors),
```
Comment (edleeks87) [inline src/components/message/message.component.js]: Might be good to get some comments on these propTypes so they're surfaced in the storybook prop types table

Comment (dawidzarzycki) [reply src/components/message/message.component.js]: Updated :)

**Pull request merged**